coding_systems/bnf/tests/test_import_data.py

```diff
@@ -162,79 +158,93 @@ def test_import_data_too_many_csv_files(tmp_path):
         )
 
 
-def test_import_data(
-    settings, coding_systems_database_tmp_dir, mock_bnf_import_data_path
-):
-    cs_release_count = CodingSystemRelease.objects.count()
+class BNFDynamicDatabaseTestCase(DynamicDatabaseTestCase):
+    # The BNF tests in this module are a special case where they use fixtures
+    # to write temporary files, instead of having the fixtures as repository
+    # files.
+    # This is set to autouse as all the tests using this class,
+    # use import data from a fixture.
+    # It could be generalised by using the "request" fixture and
+    # request.getfixturevalue(), but we don't need to right now.
+    @pytest.fixture(autouse=True)
+    @pytest.mark.usefixtures("mock_bnf_import_data_path")
+    def _set_import_data_from_fixture(self, mock_bnf_import_data_path):
+        self.import_data_path = mock_bnf_import_data_path
```

Comment:
question: We could simplify this by writing the import data out to a file, as it should then be usable like the other import data for other tests.

The data is used in `import_data()` and, in most cases, exists in the repository as files; it's possible that the reason it wasn't done here, is because a later author chose not to use the same style.

Is this worth doing, or just leave it as is? It's not a terribly complex special case.

Reply:
If it's a stylistic choice then making them consistent is a nice opportunity to reduce complexity (although adding more work to this PR). Could refactoring this be deferred to its own issue?

Reply:
> Could refactoring this be deferred to its own issue?

Yes, I think that would be fine.

There is an opportunity to get rid of this extra setup. But the extra setup is not too complex: there is an existing pytest fixture that creates the data, and that gets assigned to an attribute on the specific test class. All this happens in the same Python module.

It's just one of the slight inconsistencies I found when trying to generalise the tests using dynamically created databases.